backend/src/models/analysis.py

```diff
@@ -106,3 +106,25 @@ def units_to_annotate(self):
                     })
 
         return units
+
+    def find_discussion_post(self, discussion_post_id):
+        """ Finds a specific discussion post in an analysis by the discussion post id """
+        for discussion in self.discussions:
+            if discussion['post_id'] == discussion_post_id:
+                return discussion
+
+        return None
+
+    def find_authored_discussion_post(self, discussion_post_id, client_id):
```

Comment:
Would like to see a unit test added for the model that includes the case where it does not find a discussion post with that ID authored by that ID, since that is the case we are verifying.

Reply:
Added new units tests to test the functionality of the two new model functions for the Analysis model:

test_find_dicussion_post
test_find_dicussion_post_not_found
test_find_authored_discussion_post
test_find_authored_discussion_post_failure_missing_post
test_find_authored_discussion_post_author_mismatch

And new integration tests for the analysis router in the event of failures:

test_update_post_in_analysis_author_mismatch
test_handle_delete_post_not_existing_in_analysis

Reply:
Tests look great!